feature: resizable budget columns

samhpickering opened the pull request:
GitHub Issue (if applicable): #XXX

Trello Link (if applicable):

**Explanation of Bugfix/Feature/Modification:**
At small screen resolutions the number columns in the budget can be crushed. This feature allows clicking on the column headers to force the columns wider at the expense of category name width. It will remember the width set between visits.

![resizable-columns](https://user-images.githubusercontent.com/83871143/130369890-3761aa6e-e5b9-4302-b298-bdea5f65e2eb.png)
![resizable-columns-2](https://user-images.githubusercontent.com/83871143/130369892-7573f95a-00cf-47ab-9b12-f95530d5c584.png)

joshmadewell commented on src/extension/features/budget/resizable-columns/settings.js:
```diff
@@ -0,0 +1,8 @@
+  default: false,
+  section: 'budget',
+  title: 'Resizable Columns',
+  description: `Allows clicking/right-clicking the budget table headers to grow/shrink the width of the number columns. Middle-click to reset.`,
```
I want to say a large majority of OSX users (myself) use a trackpad, we should find a trackpad-friendly way to implement this feature.

StephenBrown2 replied:
Two-finger click?

joshmadewell replied:
two-finger click is how you'd shrink, middle click is used to reset but I'm not sure there's a great way to do that with a trackpad?

joshmadewell reviewed (changes requested):
I think this is a good option for the feature but we do need to come up with something a little but more trackpad friendly in the case of a reset. Honestly, maybe just in the `destroy()` method we should reset the widths and the way you reset your columns is by disabling :/

tsmith165 commented:
Would really like this feature, but at least on firefox, it does not seem to be very easy to find the correct position to drag at to resize the columns.  I think adding separator bars between the columns in the header row could solve this issue as it would show where you should be dragging from and would make it easier to see how wide you are making the columns.  

Another option that I would be fine with is just having fixed percentage column widths.  Even just being able to set the category column width to be fixed at 40% or so, and then having the rest of the columns fill in the leftover space evenly would be a huge improvement.  

Let me know if I can be of any help here!  And thanks for your work on this

samhpickering commented:
I've implemented a different interface for resizing - buttons in the header elements that show on mouseover.
![resizable-columns-3](https://user-images.githubusercontent.com/83871143/142330936-4a1f0e8e-ffe9-4895-8c99-c79fa859dbf9.png)
Left clicking the buttons will grow/shrink the column, and right clicking either button will reset the column width.

samhpickering commented:
Fixed column widths not saving - not sure how I didn't catch that one, sorry!
I've also added styling to make it clearer when the column is at its minimum width.
